docs/general/dql/selects.rst

```diff
@@ -834,6 +834,42 @@ Object property can also be addressed in the :ref:`where clause
     +------------+----------------------------------------------------------------------+
     SELECT 1 row in set (... sec)
 
+For selecting records having values with empty or unassigned objects, like::
+
+    cr> insert into locations (id, name, position, kind, inhabitants) values
+    ... (17, 'Voidspace 2', 0, 'Milliways', {});
+    INSERT OK, 1 row affected (... sec)
+
+.. Hidden: refresh locations
+
+    cr> refresh table locations;
+    REFRESH OK, 1 row affected (... sec)
+
+a suitable expression to match for both ``NULL`` values and empty objects, is::
+
+    cr> select count(*) from locations where
+    ... inhabitants is null or
+    ... inhabitants = {};
+    +----------+
+    | count(*) |
+    +----------+
+    |       13 |
+    +----------+
+    SELECT 1 row in set (... sec)
+
+For determining the number of items in an object, you can use the
+:ref:`object_keys <scalar-object_keys>` and :ref:`array_length
+<scalar-array_length>` scalar functions::
+
+    cr> select count(*) from locations where
+    ... array_length(object_keys(inhabitants), 1) = 2;
+    +----------+
+    | count(*) |
+    +----------+
+    |        1 |
+    +----------+
+    SELECT 1 row in set (... sec)
+
```

Comment:
Shall we maybe drop the new rows in a hidden statement so later result sets must not me adjusted?

Reply:
I've amended the patch to do it like you suggested.